BUG: Histogramdd breaks on big arrays in Windows

navpreetnp7 opened the pull request:
Resolved the issue with line change from int to np.intp in numpy/numpy/lib/histograms.py
New test function was added called test_big_arrays on numpy/lib/tests/test_histograms.py

<!--         ----------------------------------------------------------------
                MAKE SURE YOUR PR GETS THE ATTENTION IT DESERVES!
                ----------------------------------------------------------------

*  FORMAT IT RIGHT:
      https://www.numpy.org/devdocs/dev/development_workflow.html#writing-the-commit-message

*  IF IT'S A NEW FEATURE OR API CHANGE, TEST THE WATERS:
      https://www.numpy.org/devdocs/dev/development_workflow.html#get-the-mailing-list-s-opinion

*  HIT ALL THE GUIDELINES:
      https://numpy.org/devdocs/dev/index.html#guidelines

*  WHAT TO DO IF WE HAVEN'T GOTTEN BACK TO YOU:
      https://www.numpy.org/devdocs/dev/development_workflow.html#getting-your-pr-reviewed
-->

Resolved #22288 
Pydata Sprint NYC 2022

navpreetnp7 committed:
BUG: Histogramdd breaks on big arrays in Windows

Resolved the issue with line change from int to np.intp in numpy/numpy/lib/histograms.py

ganesh-k13 commented on numpy/lib/histograms.py:
```diff
@@ -970,7 +970,8 @@ def histogramdd(sample, bins=10, range=None, density=None, weights=None):
         N, D = sample.shape
 
-    nbin = np.empty(D, int)
+    #nbin = np.empty(D, int)
```
Can you remove this line

navpreetnp7 replied:
Yes, and committed.

navpreetnp7 committed:
BUG: Histogramdd breaks on big arrays in Windows

Resolved the issue with line change from int to np.intp in numpy/numpy/lib/histograms.py

seberg commented on numpy/lib/tests/test_histograms.py:
```diff
@@ -397,6 +397,14 @@ def test_histogram_bin_edges(self):
         edges = histogram_bin_edges(arr, bins='auto', range=(0, 1))
         assert_array_equal(edges, e)
 
+    def test_big_arrays(self):
```
Thanks.  We need to mark this for large memory use, have a look for `@requires_memory`.  I suspect it should also be marked with `@pytest.mark.slow` to skip on most runs.

seberg commented:
There are two binary files that got accidentally commit, please delete them.

navpreetnp7 committed:
Removed the binary files

seberg commented:
@navpreetnp7 would be great to finish this up, since branching will be in a week or two probably.  Could you add that `@requires_memory` and `@pytest.mark.slow` mark to the test?

CI/tests are currently failing because the test requires a huge amount of memory to run successfully.

navpreetnp7 committed:
Update test_histograms.py

navpreetnp7 committed:
Update test_histograms.py

navpreetnp7 commented:
Yeah I just finished it up. Sorry for the delay

seberg commented on numpy/lib/tests/test_histograms.py:
```diff
@@ -397,6 +398,16 @@ def test_histogram_bin_edges(self):
         edges = histogram_bin_edges(arr, bins='auto', range=(0, 1))
         assert_array_equal(edges, e)
 
+    @requires_memory(free_bytes=100000000 * 3)
```
Thanks.  Lets refine this a little bit.  If you check `sample.nbytes`, you will see that it is 8 times larger.   Also the result array (with the current choices) has a significant size.  You could just reduce that size by a lot, even then it probably makes sense to round up a little bit.
(This is a sanity check that we should be good after all.  The decorator also catches `MemoryError` so we should be fine, but I would just round up generously.)

navpreetnp7 replied:
Sure. That makes sense. I made it 100 times larger

seberg replied:
OK, if I do the math, this should be about double of what is needed, and 10 GiB are not forbidding these days, so this should be fine.

navpreetnp7 committed:
Update test_histograms.py

seberg commented:
Thanks @navpreetnp7.
